app/controllers/sessions_controller.rb

```diff
@@ -21,6 +24,48 @@ def mfa_create
     end
   end
 
+  def webauthn_authentication_options
+    user = User.find_by_name(session[:mfa_user])
```

Comment:
I see this is borrowing from existing code, which is understandable. That existing code and this would benefit from being specific about how the lookup is done and only lookup by handle as this is what is stored in `session[:mfa_user] = @user.handle`.

`User.find_by_name` searches by email first and only if that fails, does it lookup by handle:

```rb
def self.find_by_name(name)
  find_by(email: name) || find_by(handle: name)
end
```

Alternatively, store user id `session[:mfa_user] = @user.id` and lookup by `User.find(session[:mfa_user])`.

Reply:
Good call.
Thank you!